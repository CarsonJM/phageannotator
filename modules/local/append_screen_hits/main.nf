process APPEND_SCREEN_HITS {
    tag "$meta.id"
    label 'process_low'

    conda "conda-forge::biopython=1.78 conda-forge::pandas=1.3.5"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'quay.io/biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    path reference_fasta
    tuple val(meta), path(mash_screen)
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("*.fasta_w_mash_hits.fna.gz") , emit: fasta_w_mash_hits
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    append_screen_hits.py \\
        --reference_fasta $reference_fasta \\
        --mash_screen_results $mash_screen \\
        --assembly_fasta $fasta \\
        --prefix $prefix \\
        --output ${prefix}.fasta_w_mash_hits.fna

    gzip ${prefix}.fasta_w_mash_hits.fna

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        biopython: \$(echo \$(biopython_version.py 2>&1))
        pandas: \$(echo \$(pandas_version.py 2>&1))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        : \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//' ))
    END_VERSIONS
    """
}
