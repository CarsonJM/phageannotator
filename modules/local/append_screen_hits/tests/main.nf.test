nextflow_process {

    name "Test Process APPEND_SCREEN_HITS"
    script "modules/local/append_screen_hits/main.nf"
    process "APPEND_SCREEN_HITS"
    tag "append_screen_hits"
    tag "modules_local"

    test("Test append_screen_hits") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = file(params.test_data['modules_local']['reference_fasta_gz'], checkIfExists: true)
                input[1] = [ [id: 'test'], [file(params.test_data['modules_local']['mash_screen_results_tsv'], checkIfExists: true)] ]
                input[2] = [ [id: 'test'], [file(params.test_data['modules_local']['assembly_fasta_gz'], checkIfExists: true)] ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
