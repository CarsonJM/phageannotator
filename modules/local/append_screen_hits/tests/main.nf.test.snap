{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_mash_hits.fna.gz:md5,9352e3ea473792e7ed2ff781c75cae8f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8e9e3bd75f037dbd06521415d76ab42c"
                ],
                "fasta_w_mash_hits": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_mash_hits.fna.gz:md5,9352e3ea473792e7ed2ff781c75cae8f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8e9e3bd75f037dbd06521415d76ab42c"
                ]
            }
        ],
        "timestamp": "2023-08-03T22:39:09+0000"
    }
}